### PR TITLE
Bug 1948022: Add on-prem namespaces to relatedObjects

### PR DESCRIPTION
**Pull request by cybertron**
Previously the kni namespace was explicitly listed in must-gather,
but when I went to add the other on-prem ones it was pointed out
that this should probably be done with relatedObjects. This adds
the namespaces here so we don't need anything in must-gather itself.

**- Description for the changelog**
Added on-prem service namespaces to relatedObjects so they will be included in must-gather.

### CODE REVIEW

**Comment by cybertron** [comment]
/test e2e-openstack
/test e2e-ovirt
/test e2e-vsphere
/cc @mandre @Gal-Zaidman @jcpowermac @patrickdillon

As requested on the must-gather review doing the same.

**Review by mandre — COMMENTED** [review]
I'm a bit unfamiliar with this part of the code, but LGTM on the principle.

So, once this merge, you could update https://github.com/openshift/must-gather/pull/209/ to remove the `openshift-kni-infra` namespace from the named resource collection?

**Comment by cybertron** [comment]
> So, once this merge, you could update [openshift/must-gather#209](https://github.com/openshift/must-gather/pull/209) to remove the `openshift-kni-infra` namespace from the named resource collection?

Yes, we would no longer need anything specific to the on-prem platforms in must-gather. The only reason I initially added the kni namespace there was that I didn't know about this. :-)

**Comment by cybertron** [comment]
/retest

Note that although the vsphere job failed, it did capture logs this time so it confirms that this is working.

File: pkg/operator/status.go
Comment (kikisdeliveryservice) [inline pkg/operator/status.go:75]: could you add a comment above the new additions just to memorialize what this is for?

Comment (cybertron) [reply pkg/operator/status.go:76]: Done

**Comment by cybertron** [comment]
Both of the remaining failed jobs look pretty red in general. They don't seem to be related to this patch (not sure how they could be).

**Comment by cybertron** [comment]
@Gal-Zaidman @mandre Can I get another lgtm on this? I lost it when I added the comment. Thanks.

**Review by mandre — COMMENTED** [review]
/lgtm

**Review by kikisdeliveryservice — APPROVED** [review]
Double-checked and this seems to be working correctly.

**Comment by mandre** [comment]
/retest
Why isn't prow restarting e2e-agnostic-upgrade?

**Comment by kikisdeliveryservice** [comment]
@cybertron Do you still want this backported?

**Comment by cybertron** [comment]
Definitely. It's a real hassle debugging on-prem network issues without it. Let's see if I remember the cherry-pick syntax...

/cherry-pick release-4.7

**Pull request merged**
